config/default/tasks.js

```diff
@@ -0,0 +1,334 @@
+      dueDate: function (event, contact, report) {
+        //next visit date
+        return getDateISOLocal(getField(report, 't_pregnancy_follow_up_date'));
+      }
+    }
+    ]
+  },
+  {
+    icon: "icon-pregnancy-danger",
+    title: "task.anc.pregnancy_danger_sign_followup.title",
+    appliesTo: "reports",
+    appliesToType: ["pregnancy", "pregnancy_home_visit", "pregnancy_danger_sign", "pregnancy_danger_sign_follow_up"],
+    appliesIf: function (contact, report) {
+      return getField(report, 't_danger_signs_referral_follow_up') === 'yes' && isAlive(contact);
+    },
+    resolvedIf: function (contact, report, event, dueDate) {
+      //(refused or migrated) and clear tasks 
+      if (isPregnancyTaskMuted(contact)) return true;
+      const startTime = Math.max(addDays(dueDate, -event.start).getTime(), report.reported_date + 1);//+1 so that source ds_follow_up does not resolve itself
+      const endTime = addDays(dueDate, event.end + 1).getTime();
+      return isFormArraySubmittedInWindow(contact.reports, ['pregnancy_danger_sign_follow_up'], startTime, endTime);
+    },
+    actions: [
+      {
+        type: "report",
+        form: "pregnancy_danger_sign_follow_up",
+        //label: "Pregnancy follow up",
+        modifyContent: function (content, contact, report) {
+          content.pregnancy_uuid = report._id;
+        }
+      }
+    ],
+    events: [
+      {
+        id: "pregnancy-danger-sign-follow-up",
+        start: 3,
+        end: 7,
+        dueDate: function (event, contact, report) {
+          return getDateISOLocal(getField(report, 't_danger_signs_referral_follow_up_date'));
+        }
+      }
+    ]
+  },
+
+  {
+    icon: "icon-mother-child",
+    title: "task.anc.delivery.title",
+    appliesTo: "reports",
+    appliesToType: ["pregnancy"],
+    appliesIf: function (contact, report) {
+      const lmpDate = getMostRecentLMPDateForPregnancy(contact, report); //only for known LMP
+      return lmpDate && addDays(lmpDate, 337) >= today && isAlive(contact);
+    },
+    resolvedIf: function (contact, report, event, dueDate) {
+      //miscarriage or abortion
+      if (getRecentANCVisitWithEvent(contact, report, 'abortion') || getRecentANCVisitWithEvent(contact, report, 'miscarriage')) return true;
+
+      //(refused or migrated) and clear tasks 
+      if (isPregnancyTaskMuted(contact)) return true;
+      const startTime = Math.max(addDays(dueDate, -event.start).getTime(), report.reported_date);
+      const endTime = addDays(dueDate, event.end + 1).getTime();
+      return isFormArraySubmittedInWindow(contact.reports, ['delivery'], startTime, endTime);
+    },
+    actions: [
+      {
+        type: "report",
+        form: "delivery",
+        //label: "Delivery",
+        modifyContent: function (content, contact, report) {
+          content.pregnancy_uuid = report._id;
+        }
+      }
+    ],
+    events: [
+      {
+        id: "delivery-reminder",
+        start: 28, //due - 4 weeks 
```

Comment:
In the spirit of writing self documenting code, you could do something like `start: 4 * 7` and `end: 6 * 7` so the number of week is obvious and the "what this is" comment is unnecessary.

Reply:
Done.